### PR TITLE
Update keepalive.md

### PR DESCRIPTION
**Pull request by yashykt**
Update keepalive.md with feedback from #25722

### CODE REVIEW

File: doc/keepalive.md [hunk 1]
```diff
@@ -4,25 +4,27 @@ The keepalive ping is a way to check if a channel is currently working by sendin
...
 
 * **GRPC_ARG_KEEPALIVE_TIME_MS**
   * This channel argument controls the period (in milliseconds) after which a keepalive ping is sent on the transport.
 * **GRPC_ARG_KEEPALIVE_TIMEOUT_MS**
   * This channel argument controls the amount of time (in milliseconds) the sender of the keepalive ping waits for an acknowledgement. If it does not receive an acknowledgment within this time, it will close the connection.
-
-The above two channel arguments should be sufficient for most users, but the following arguments can also be useful in certain use cases.
-
+* **GRPC_ARG_HTTP2_MAX_PINGS_WITHOUT_DATA**
+  * This channel argument controls the maximum number of pings that can be sent when there is no data/header frame to be sent. GRPC Core will not continue sending pings if we run over the limit. Setting it to 0 allows sending pings without such a restriction. (Note that this is an unfortunate setting that does not agree with [A8-client-side-keepalive.md](https://github.com/grpc/proposal/blob/master/A8-client-side-keepalive.md). There should ideally be no such restriction on the keepalive ping and we plan to deprecate it in the future.)
```
Comment (drfloob) [inline doc/keepalive.md:14]: I think we should use consistent case for "gRPC Core". It is called "GRPC Core" in the second sentence here, and "gRPC core" in another spot.

File: doc/keepalive.md [hunk 2]
```diff
@@ -4,25 +4,27 @@ The keepalive ping is a way to check if a channel is currently working by sendin
 
 This guide documents the knobs within gRPC core to control the current behavior of the keepalive ping.
 
-The keepalive ping is controlled by two important channel arguments -
+The send-side keepalive ping in core is controlled by the following channel arguments -
```
Comment (drfloob) [inline doc/keepalive.md:7]: I believe this is labeled "client-side" in the IMPORTANT NOTE below. Should we choose one or the other term for consistency? Or is there a further distinction between them?

File: doc/keepalive.md [hunk 3]
```diff
@@ -4,25 +4,27 @@ The keepalive ping is a way to check if a channel is currently working by sendin
...
+
 * **GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS**
   * If there are no data/header frames being sent on the transport, this channel argument on the server side controls the minimum time (in milliseconds) that gRPC Core would expect between receiving successive pings. If the time between successive pings is less that than this time, then the ping will be considered a bad ping from the peer. Such a ping counts as a ‘ping strike’.
 On the client side, this does not have any effect.
 * **GRPC_ARG_HTTP2_MAX_PING_STRIKES**
   * This arg controls the maximum number of bad pings that the server will tolerate before sending an HTTP2 GOAWAY frame and closing the transport. Setting it to 0 allows the server to accept any number of bad pings.
 
+**IMPORTANT NOTE** - For keepalive to work properly and as intended, all of the above channel arguments should be configured appropriately. The client-side keepalive settings should also be in `agreement` with the server-side settings. If a client sends pings more often than the server is willing to accept, the connection will be terminated with a GOAWAY frame with "too_many_pings" as the debug data.
```
Comment (drfloob) [inline doc/keepalive.md:26]: Is `agreement` intended to be a code snippet? Did you mean to quote it instead, maybe? I'm not sure it needs to be quoted.

File: doc/keepalive.md [hunk 4]
```diff
@@ -4,25 +4,27 @@ The keepalive ping is a way to check if a channel is currently working by sendin
...
-
+* **GRPC_ARG_HTTP2_MAX_PINGS_WITHOUT_DATA**
+  * This channel argument controls the maximum number of pings that can be sent when there is no data/header frame to be sent. GRPC Core will not continue sending pings if we run over the limit. Setting it to 0 allows sending pings without such a restriction. (Note that this is an unfortunate setting that does not agree with [A8-client-side-keepalive.md](https://github.com/grpc/proposal/blob/master/A8-client-side-keepalive.md). There should ideally be no such restriction on the keepalive ping and we plan to deprecate it in the future.)
 * **GRPC_ARG_KEEPALIVE_PERMIT_WITHOUT_CALLS**
   * This channel argument if set to 1 (0 : false; 1 : true), allows keepalive pings to be sent even if there are no calls in flight.
-* **GRPC_ARG_HTTP2_MAX_PINGS_WITHOUT_DATA**
-  * This channel argument controls the maximum number of pings that can be sent when there is no data/header frame to be sent. GRPC Core will not continue sending pings if we run over the limit. Setting it to 0 allows sending pings without such a restriction.
+
+On the server-side, the following channel arguments need to be configured -
```
Comment (drfloob) [inline doc/keepalive.md:18]: Suggest either "client-side/server-side" or "send-side/receive-side".

**Review by drfloob — APPROVED** [review]
Sorry this PR went unnoticed. Thanks for updating the doc! LGTM, I left a few small comments on word choice / consistency.

File: doc/keepalive.md (hunk 2 above)
Comment (yashykt) [reply doc/keepalive.md:7]: "client-side" actually differs from "send-side" here. For example, these args GRPC_ARG_KEEPALIVE_TIME_MS and others are also valid on the server side, while the "server-side" args are args that only affect servers.

File: doc/keepalive.md (hunk 4 above)
Comment (yashykt) [reply doc/keepalive.md:18]: (Same as earlier comment) Maybe I need some better wording to make it clear.

File: doc/keepalive.md (hunk 3 above)
Comment (yashykt) [reply doc/keepalive.md]: Removed the quote

File: doc/keepalive.md (hunk 1 above)
Comment (yashykt) [reply doc/keepalive.md]: Done

File: doc/keepalive.md (hunk 2 above)
Comment (drfloob) [reply doc/keepalive.md:7]: I misunderstood, thanks for clarifying. Send-side sounds fine.

File: doc/keepalive.md (hunk 4 above)
Comment (drfloob) [reply doc/keepalive.md:18]: Covered above. I think it's probably fine as is.

File: doc/keepalive.md (hunk 2 above)
Comment (yashykt) [reply doc/keepalive.md]: I actually removed "send-side" since it is confusing that way

**Pull request merged**